src/ops/cli/ssh.py

```diff
@@ -132,18 +133,28 @@ def run(self, args):
         if len(hosts) <= args.index:
             group = args.role
             hosts = self.ansible_inventory.get_hosts(group)
-            if len(hosts) <= args.index:
-                return
+            if not hosts:
+                display("No host found in inventory, using provided name %s" % (args.role), color="purple", stderr=True)
 
         display("Expression %s matched hosts (max 10): " % group, stderr=True)
         host_names = [host.name for host in hosts]
         for name in host_names[:10]:
             display(name, color='blue')
 
-        host = self.ansible_inventory.get_host(host_names[args.index])
-
-        ssh_host = host.vars.get('ansible_ssh_host') or host.name
-
+        host = None
+        if host_names:
+            if args.index < len(host_names):
+                host = self.ansible_inventory.get_host(host_names[args.index])
+            else:
+                display("Index out of bounds for %s" % (group), color="red", stderr=True)
+                return
+        if host:
+            ssh_host = host.vars.get('ansible_ssh_host') or host.name
+        else:
+            # no host found in inventory, use the role provided
+            bastion = self.ansible_inventory.get_hosts('bastion')[0].vars.get('ec2_ip_address')
+            ssh_host = '{}--{}'.format(bastion, args.role)
```

Comment:
More of a cosmetic change:

Instead of:
```python
ssh_host = '{}--{}'.format(bastion, args.role)
host = Host(name=args.role)
```

what about going with:
```python
host = Host(name=args.role)
ssh_host = '{}--{}'.format(bastion, host.name)
```
?

Reply:
better indeed :)